botocore/endpoint.py

```diff
@@ -152,16 +130,13 @@ def _create_request_object(self, request_dict):
                              headers=headers)
         return request
 
-    def prepare_request(self, request, signer):
-        if signer is not None:
-            with self._lock:
```

Comment:
Will the lock be needed in the RequestSigner class? I can see the case where a client is shared across multiple threads.